packages/multiselect/src/react/utils.ts

```diff
@@ -0,0 +1,33 @@
+import { Option } from './types'
+
+export function arraysAreEqual<T>(a: T[] = [], b: T[] = []) {
+  return a.length === b.length && a.every(item => b.includes(item))
+}
+
+/* eslint-disable @typescript-eslint/no-empty-function */
+export function noop() {}
+/* eslint-enable @typescript-eslint/no-empty-function */
+
+export function simpleTextFilter(term = '', options: Option[]) {
+  if (term.length === 0) return options
+
+  const needle = term.toLowerCase()
+
+  return options.filter(option => {
+    const { label, value } = option
+
+    return (
+      label.toLowerCase().includes(term) || value.toLowerCase().includes(needle)
+    )
+  })
+}
+
+/* eslint-disable no-prototype-builtins */
+export function switchcase<T = unknown>(
+  cases: Record<string, T>,
+  defaultCase: T,
+  key: string
+): T {
+  return cases.hasOwnProperty(key) ? cases[key] : defaultCase
+}
```

Comment:
What was the reason for creating this?  I'm assuming it was the only way to get a { [str]: Fn } may for selecting a Fn dynamically to TS compile.

Reply:
i wanted a switchcase fn that used objects instead of using native switch cases. i find it easier to reason about than a native switch because there is no chance of fall-through and there is a require defaultCase. Also...it's probably faster because it's lookup by key...but that's an assumption